sources/account_graph.move

```diff
@@ -100,26 +111,26 @@ module account_graph::account_graph {
 
     /// Remove a relationship as edge, where source is the sender,
     /// fail if node doesn't exist. Property of the relationship
-    /// will be removed if exists. If `remove_empty_vec` is `true`,
-    /// the adjaceny list will be removed when it is empty.
+    /// will be removed if exists. It also unsets relationship
+    /// property in `relationship_props` if there is any.
     public fun remove_relationship<AccountProps: copy + drop + store, RelationshipProps: copy + drop + store>(
         self: &mut AccountGraph<AccountProps, RelationshipProps>,
         target: address,
-        remove_empty_vec: bool,
         ctx: &mut TxContext,
     ): (address, Option<RelationshipProps>) {
         let source = sender(ctx);
         let adj_lists = table::borrow_mut(&mut self.relationships, source);
         vec_set::remove(adj_lists, &target);
-        if (remove_empty_vec && vec_set::size(adj_lists) == 0) {
+        if (vec_set::size(adj_lists) == 0) {
             table::remove(&mut self.relationships, source);
```

Comment:
I think you will also need to remove the `account_props` here if any exist (and also possibly check that the vertex exists in the graph when the account prop is added).

Reply:
This was intentionally left unchecked. Conceptually, ALL accounts exist in this graph as vertices, not just those that have relationships with other accounts. By definition, no one can add or remove a vertex. Owners have the ability to set optional props for their accounts.

This is to optimize the add relationship flow, such that it doesn't require target accounts to pre-exist as entries. It also avoids the ambiguity of who can add or remove an account from the graph and what happens to incoming relationships when an account is removed.